src/Common/Core/Impl/Threading/MainThreadExtensions.cs

```diff
@@ -7,5 +7,8 @@ namespace Microsoft.Common.Core.Threading {
     public static class MainThreadExtensions {
         public static MainThreadAwaitable SwitchToAsync(this IMainThread mainThread, CancellationToken cancellationToken = default(CancellationToken))
             => new MainThreadAwaitable(mainThread, cancellationToken);
+
+        public static bool CheckAccess(this IMainThread mainThread)
```

Comment:
Should this be something like `IsOnMainThread`? It's not clear what access is being checked here...